cfgv.py

```diff
@@ -162,6 +162,18 @@ def _no_additional_keys_check(self, dct):
         )
 
 
+def _warn_additional_keys_check(self, dct):
+    extra = sorted(set(dct) - set(self.keys))
+    if extra:
+        self.logger.warning(
```

Comment:
instead of boxing into the `logger` api, let's make this just a callback function, the caller can decide whether they want to use `logging` or whatever

Reply:
Ok, I see, I will add this change. 
This problem was already tried to solve in (too) many ways :), but this is ok I have got new experiences.